File: library/src/main/java/cc/shinichi/library/tool/file/SingleMediaScanner.java
package cc.shinichi.library.tool.file;

import android.content.Context;
import android.media.MediaScannerConnection;
import android.net.Uri;

/**
 * @author 工藤
 * @email qinglingou@gmail.com
 * create at 2018/5/4  16:50
 * description:媒体扫描
 */
public class SingleMediaScanner implements MediaScannerConnection.MediaScannerConnectionClient {

    private final MediaScannerConnection mMs;
    private final String path;
    private final ScanListener listener;

    public SingleMediaScanner(Context context, String path, ScanListener l) {
        this.path = path;
        this.listener = l;
        this.mMs = new MediaScannerConnection(context, this);
        this.mMs.connect();
    }

    @Override
    public void onMediaScannerConnected() {
        mMs.scanFile(path, null);
    }

    @Override
    public void onScanCompleted(String path, Uri uri) {
        mMs.disconnect();
        if (listener != null) {
            listener.onScanFinish();
        }
    }

    public interface ScanListener {
        void onScanFinish();
    }
}

File: library/src/main/java/cc/shinichi/library/tool/file/SingleMediaScanner.kt
package cc.shinichi.library.tool.file

import android.content.Context
import android.media.MediaScannerConnection
import android.media.MediaScannerConnection.MediaScannerConnectionClient
import android.net.Uri

/**
 * @author 工藤
 * @email qinglingou@gmail.com
 * create at 2018/5/4  16:50
 * description:媒体扫描
 */
class SingleMediaScanner(context: Context?, private val path: String, private val listener: ScanListener?) :
    MediaScannerConnectionClient {

    private val mMs: MediaScannerConnection = MediaScannerConnection(context, this)

    override fun onMediaScannerConnected() {
        mMs.scanFile(path, null)
    }

    override fun onScanCompleted(path: String, uri: Uri) {
        mMs.disconnect()
        listener?.onScanFinish()
    }

    interface ScanListener {
        fun onScanFinish()
    }

    init {
        mMs.connect()
    }
}